ui/jobs/components/sge.ts

```diff
@@ -76,10 +82,25 @@ export class SGEComponent extends BaseComponent {
   }
 
   override onUseAbility(id: string): void {
-    if (id === kAbility.LucidDreaming)
-      this.lucidDream.duration = 60;
-    else if (id === kAbility.Rhizomata)
-      this.rhizomata.duration = 90;
+    switch (id) {
+      case kAbility.Phlegma:
+      case kAbility.Phlegma2:
+      case kAbility.Phlegma3:
+        if (!this.multipleCall) { // Avoid multiple call in AOE
```

Comment:
I think you'd have to pass in the entire matches here, but you can use the `matches.targetIndex` field to only trigger on the first target of the aoe rather than using a countdown.  I think this is fine for now, but maybe something to fix up if you wanted?

Reply:
@MaikoTan said he is trying to import a t
throttle function in util to simplify this. I'm not sure which means is better? I've not thought about how to use targetIndex to do this

Reply:
Every enemy hit by an ability gets a different target index.  You can see this in logs if you look at aoe ability lines.  So you can check the match and only consider targetIndex of 0 when doing things like this.

Reply:
Echoring came to me and asked how to restrict a function so can only be invoked once in some secondes, so I answered "throttle function" 😂.
Looks like in this case it can be solved with the `targetIndex` that mentioned above. That's great !